app/models/order_detail.rb

```diff
@@ -345,7 +345,7 @@ class << self
     search = search.joins(:journal) if action.to_sym == :journal_date
 
     # If we're searching on fulfilled_at, ignore any order details that don't have a fulfilled date
-    search = search.where("fulfilled_at IS NOT NULL") if action.to_sym == :fulfilled_at
+    search = search.where("#{action} IS NOT NULL") if action.to_sym == :fulfilled_at || :reconciled_at
```

Comment:
I think this actually compiles down to `if (action.to_sym == :fulfilled_at) || :reconciled_at`, so it would always be true.

It's whitelisted, so it's not a real SQL injection possiblity, but it would be nice to just avoid that error altogether, either have this as two separate lines or I think we could do `search.where.not(action =>nil)`.

Part of me almost wants to further refactor this scope to use a `case` statement (as a possible first step to a further refactoring).